File: app/src/main/java/com/dumuzeyn/mp3player/ArtistsMenuRenderer.java
package com.dumuzeyn.mp3player;

import java.util.ArrayList;
import java.util.Map;

final class ArtistsMenuRenderer extends TrackGroupMenuRenderer {
    ArtistsMenuRenderer(MainActivityCore host) {
        super(host);
    }

    @Override
    Map<String, ArrayList<Track>> groupedTracks() {
        return host.libraryState.homeContent.artistTracks;
    }

    @Override
    String unknownGroupName() {
        return host.tr("Unknown artist", "Неизвестный исполнитель");
    }

    @Override
    int cardOpacity() {
        return host.appearanceState.artistCardOpacity;
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/ArtistsMenuRenderer.kt
package com.dumuzeyn.mp3player

internal class ArtistsMenuRenderer(host: MainActivityCore) : TrackGroupMenuRenderer(host) {
    override fun groupedTracks(): Map<String, ArrayList<Track>> =
        host.libraryState.homeContent.artistTracks

    override fun unknownGroupName(): String =
        host.tr("Unknown artist", "Неизвестный исполнитель")

    override fun cardOpacity(): Int = host.appearanceState.artistCardOpacity
}
